### ISSUE TITLE
Arithmetic overflow in SDL samples

### ISSUE THREAD

**Issue by vlazar** [crystal-lang/crystal#8351]
All SDL samples are raising `Unhandled exception: Arithmetic overflow (OverflowError)`.

```
$ crystal samples/sdl/fire.cr
Unhandled exception: Arithmetic overflow (OverflowError)
  from samples/sdl/fire.cr:284:19 in 'put_pixel'
  from samples/sdl/fire.cr:452:5 in '__crystal_main'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:86:7 in 'main'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:106:3 in 'main'
```

```
$ crystal samples/sdl/raytracer.cr
Unhandled exception: Arithmetic overflow (OverflowError)
  from samples/sdl/sdl/surface.cr:27:57 in '[]='
  from samples/sdl/raytracer.cr:196:7 in 'render'
  from samples/sdl/raytracer.cr:236:10 in '__crystal_main'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:86:7 in 'main'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:106:3 in 'main'
```

```
$ crystal samples/sdl/tv.cr
Rects: 234
Unhandled exception: Arithmetic overflow (OverflowError)
  from samples/sdl/sdl/surface.cr:27:57 in '[]='
  from samples/sdl/sdl/surface.cr:31:5 in '[]='
  from samples/sdl/tv.cr:140:11 in '__crystal_main'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:97:5 in 'main_user_code'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:86:7 in 'main'
  from /usr/local/Cellar/crystal/0.31.1/src/crystal/main.cr:106:3 in 'main'
```

All examples in https://github.com/asterite/crystal_sdl2_examples with SDL2 work just fine for me BTW.

<details>
  <summary>Crystal 0.31.1 on MacOS 10.13.6</summary>

```
$ crystal --version
Crystal 0.31.1 (2019-10-02)

LLVM: 8.0.1
Default target: x86_64-apple-macosx
```
</details>

<details>
  <summary>sdl: stable 1.2.15</summary>

```
$ brew info sdl
sdl: stable 1.2.15 (bottled), HEAD
Low-level access to audio, keyboard, mouse, joystick and graphics
https://www.libsdl.org/
/usr/local/Cellar/sdl/1.2.15_1 (225 files, 1.4MB) *
  Poured from bottle on 2019-10-15 at 17:32:48
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/sdl.rb
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 5,175 (30 days), 16,102 (90 days), 74,318 (365 days)
install_on_request: 2,178 (30 days), 6,984 (90 days), 33,221 (365 days)
```
</details>

Note

There is a bunch of unrelated warnings similar to this one `ld: warning: text-based stub file /System/Library/Frameworks//Cocoa.framework/Cocoa.tbd and library file /System/Library/Frameworks//Cocoa.framework/Cocoa are out of sync. Falling back to library file for linking.`.

Run this to fix them (taken from this [Stackoverflow answer](https://stackoverflow.com/questions/51314888/ld-warning-text-based-stub-file-are-out-of-sync-falling-back-to-library-file/55344565)):
```
export SDKROOT="/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk"
```

**Comment by asterite** [comment]
I think we should remove the SDL examples because they no longer work, regardless of the overflow error.

**Comment by megatux** [comment]
Hi, I created #10470  to fix the SDL tv and fire examples. raytracer example seems to work fine here.

**Comment by HertzDevil** [comment]
Resolved by #10470.
